drivers/serial/uart_numaker.c

```diff
@@ -0,0 +1,479 @@
+	case UART_CFG_PARITY_MARK:
+		return UART_PARITY_MARK;
+	case UART_CFG_PARITY_SPACE:
+		return UART_PARITY_SPACE;
+	case UART_CFG_PARITY_NONE:
+	default:
+		return UART_PARITY_NONE;
+	}
+}
+
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+static int uart_numaker_configure(const struct device *dev, const struct uart_config *cfg)
+{
+	const struct uart_numaker_config *config = dev->config;
+	struct uart_numaker_data *pData = dev->data;
+	int32_t databits, stopbits;
+	uint32_t parity;
+
+	databits = uart_numaker_convert_datalen(cfg->data_bits);
+	if (databits < 0) {
+		return databits;
+	}
+
+	stopbits = uart_numaker_convert_stopbit(cfg->stop_bits);
+	if (stopbits < 0) {
+		return stopbits;
+	}
+
+	if (cfg->flow_ctrl == UART_CFG_FLOW_CTRL_NONE) {
+		UART_DisableFlowCtrl(config->uart);
+	} else if (cfg->flow_ctrl == UART_CFG_FLOW_CTRL_RTS_CTS) {
+		UART_EnableFlowCtrl(config->uart);
+	} else {
+		return -ENOTSUP;
+	}
+
+	parity = uart_numaker_convert_parity(cfg->parity);
+
+	UART_SetLineConfig(config->uart, cfg->baudrate, databits, parity, stopbits);
+
+	memcpy(&pData->ucfg, cfg, sizeof(*cfg));
+
+	return 0;
+}
+
+static int uart_numaker_config_get(const struct device *dev, struct uart_config *cfg)
+{
+	struct uart_numaker_data *pData = dev->data;
+
+	memcpy(cfg, &pData->ucfg, sizeof(*cfg));
+
+	return 0;
+}
+#endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
+
+static int uart_numaker_init(const struct device *dev)
+{
+	const struct uart_numaker_config *config = dev->config;
+	struct uart_numaker_data *pData = dev->data;
+	int err = 0;
+
+	SYS_UnlockReg();
+
+#ifdef CONFIG_CLOCK_CONTROL_NUMAKER_SCC
+	struct numaker_scc_subsys scc_subsys;
+
+	memset(&scc_subsys, 0x00, sizeof(scc_subsys));
+	scc_subsys.subsys_id = NUMAKER_SCC_SUBSYS_ID_PCC;
+	scc_subsys.pcc.clk_modidx = config->clk_modidx;
+	scc_subsys.pcc.clk_src = config->clk_src;
+	scc_subsys.pcc.clk_div = config->clk_div;
+
+	/* Equivalent to CLK_EnableModuleClock() */
+	err = clock_control_on(config->clk_dev, (clock_control_subsys_t)&scc_subsys);
+	if (err != 0) {
+		goto move_exit;
+	}
+	/* Equivalent to CLK_SetModuleClock() */
+	err = clock_control_configure(config->clk_dev, (clock_control_subsys_t)&scc_subsys, NULL);
+	if (err != 0) {
+		goto move_exit;
+	}
+#else
+	/* Enable UART module clock */
+	CLK_EnableModuleClock(config->clk_modidx);
+
+	/* Select UART module clock source/divider */
+	CLK_SetModuleClock(config->clk_modidx, config->clk_src, config->clk_div);
+#endif
+
+	/* Set pinctrl for UART0 RXD and TXD */
+	/* Set multi-function pins for UART0 RXD and TXD */
+#ifdef CONFIG_PINCTRL
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err != 0) {
+		goto move_exit;
+	}
+#endif
+
+	SYS_ResetModule(config->id_rst);
+
+	UART_Open(config->uart, pData->ucfg.baudrate);
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	config->irq_config_func(dev);
+#endif
+
+move_exit:
+	SYS_LockReg();
+	return err;
+}
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+/* API implementation: fifo_fill */
```

Comment:
I would drop these comments, they are really just repeating the same information that you have in the function name itself.

Reply:
Agree with you, it's redundant.